src/app/modules/design-patterns/abstract-factory/interfaces/goalkeeper-player.interface.ts

```diff
@@ -0,0 +1,5 @@
+import { AbstractBasePlayer } from './base-player.interface';
+
+export interface IGoalKeeperPlayer extends AbstractBasePlayer {
```

Comment:
the interface `IGoalKeeperPlayer` should extends IBasePlayer(the base player contract)

Reply:
You are doing here Interface segregation